File: libs/language/src/repo/core/schema/CompositeTimelineChannel.ts
import { z } from 'zod';
import { blueIds } from '../blue-ids';
import { withTypeBlueId } from '../../../schema/annotations/typeBlueId';
import { ChannelSchema } from './Channel';

export const CompositeTimelineChannelSchema = withTypeBlueId(
  blueIds['Composite Timeline Channel']
)(
  ChannelSchema.extend({
    channels: z.array(z.string()).optional(),
  })
);

export type CompositeTimelineChannel = z.infer<
  typeof CompositeTimelineChannelSchema
>;

File: references/java-sdk/src/main/java/blue/language/types/conversation/CompositeTimelineChannel.java
package blue.language.types.conversation;

import blue.language.model.TypeBlueId;
import blue.language.types.TypeAlias;

import java.util.List;

@TypeAlias("Conversation/Composite Timeline Channel")
@TypeBlueId("HsNatiPt2YvmkWQoqtfrFCbdp75ZUBLBUkWeq84WTfnr")
public class CompositeTimelineChannel extends TimelineChannel {
    public List<String> channels;

    public CompositeTimelineChannel channels(List<String> channels) {
        this.channels = channels;
        return this;
    }
}
